### PR TITLE
SALTO-5461 - Salesforce: Propagate warnings from successful deploy results

### PR DESCRIPTION
**Pull request by tomsellek**
See ticket for an example of such a message on a successful deploy of a component

---

I made a couple of fixes to the E2E utils while I was at it.

---
_Release Notes_: 
Salesforce: info/warning messages from successfully deployed components will now be shown

---
_User Notifications_: 
N/A

### CODE REVIEW

File: packages/salesforce-adapter/src/metadata_deploy.ts [hunk 1]
```diff
@@ -308,6 +314,23 @@ const processDeployResponse = (
...
+    .filter((message) => message.problem)
+    .filter((message) => !isUnFoundDelete(message, deletionsPackageName))
+    .filter((message) => {
+      log.info('fullName: %s', message.fullName)
+      return true
+    })
+    .map((message) => ({
+      elemID: getElemIdForDeployError(message),
+      message: message.problem,
+      severity: message.problemType as SeverityLevel, // Salesforce problem types happen to match our severity levels
```
Comment (amitgoldarb) [inline packages/salesforce-adapter/src/metadata_deploy.ts:327]: This should probably be explicit. It's safer if we find an unexpected problemType or if this changes over time.

Comment (tomsellek) [reply packages/salesforce-adapter/src/metadata_deploy.ts]: Ack

File: packages/salesforce-adapter/src/metadata_deploy.ts [hunk 2]
```diff
@@ -295,11 +295,25 @@ const processDeployResponse = (
     return rawElemId
   }
 
+  const problemTypeToSeverity = (messageType: string): SeverityLevel => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/metadata_deploy.ts:298]: Why do we need this method?
Isn't the code in L325 is only relevant for warnings `Warnings`?

File: packages/salesforce-adapter/src/metadata_deploy.ts [hunk 3]
```diff
@@ -308,6 +322,23 @@ const processDeployResponse = (
...
+    .filter((message) => message.problem)
+    .filter((message) => !isUnFoundDelete(message, deletionsPackageName))
+    .filter((message) => {
+      log.info('fullName: %s', message.fullName)
+      return true
+    })
+    .map((message) => ({
+      elemID: getElemIdForDeployError(message),
+      message: message.problem,
+      severity: problemTypeToSeverity(message.problemType),
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/metadata_deploy.ts:335]: Shouldn't this be a Warning?

File: packages/salesforce-adapter/test/adapter.crud.test.ts [hunk 4]
```diff
@@ -1164,6 +1162,47 @@ describe('SalesforceAdapter CRUD', () => {
           expect(result.appliedChanges).toBeEmpty()
         })
       })
+      describe('when the successful element has warnings', () => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/adapter.crud.test.ts:1165]: Warnings should have severity `Warning` and not Info.
Do we handle Info as-well? Does the code handle it? If so, we need to change the terminology in this PR. (As it explicitly states Warning and no Info)

**Review by tamtamirr — COMMENTED** [review]
Good work!
Added some comments. Mainly confused on the scope we tackle here. Do we also handle Info?

Were you able to create a deployment that returns with warnings?
If so, let's test how the code behaves when we use `ignoreWarnings = false` in the deploy config. Will we have both warnings and errors? Should we change the severity of the Warning to Error in that use-case?

File: packages/salesforce-adapter/src/metadata_deploy.ts [hunk 5]
```diff
@@ -308,6 +322,23 @@ const processDeployResponse = (
       severity: 'Error' as SeverityLevel,
     }))
 
+  const successfulComponentWarnings = allSuccessMessages
+    .filter((message) => message.problem)
+    .filter((message) => !isUnFoundDelete(message, deletionsPackageName))
+    .filter((message) => {
+      log.info('fullName: %s', message.fullName)
```
[the remark this answers is not in the document]
Comment (tomsellek) [reply packages/salesforce-adapter/src/metadata_deploy.ts:329]: Sorry, leftover debugging stuff

File: packages/salesforce-adapter/src/metadata_deploy.ts (hunk 2 above)
Comment (tomsellek) [reply packages/salesforce-adapter/src/metadata_deploy.ts:298]: We actually saw `Info` level too (which the Salesforce docs don't mention)

File: packages/salesforce-adapter/src/metadata_deploy.ts (hunk 3 above)
Comment (tomsellek) [reply packages/salesforce-adapter/src/metadata_deploy.ts:335]: Not if it's `Info`, I think

File: packages/salesforce-adapter/test/adapter.crud.test.ts (hunk 4 above)
Comment (tomsellek) [reply packages/salesforce-adapter/test/adapter.crud.test.ts:1165]: Ack

File: packages/salesforce-adapter/src/metadata_deploy.ts [hunk 6]
```diff
@@ -308,6 +320,19 @@ const processDeployResponse = (
       severity: 'Error' as SeverityLevel,
     }))
 
+  const successfulComponentProblems = allSuccessMessages
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/metadata_deploy.ts:323]: I think we should log these problems so we can monitor this easily

**Review by tamtamirr — APPROVED** [review]
LGTM! Added a small comment to add visibility.
Approving to save cycles

**Pull request merged**